sdk/python/feast/infra/online_stores/remote.py

```diff
@@ -197,12 +427,104 @@ def _construct_online_read_api_json_request(
         )
         return req_body
 
+    def _construct_online_documents_api_json_request(
+        self,
+        table: FeatureView,
+        requested_features: Optional[List[str]] = None,
+        embedding: Optional[List[float]] = None,
+        top_k: Optional[int] = None,
+        distance_metric: Optional[str] = "L2",
+    ) -> str:
+        api_requested_features = []
+        if requested_features is not None:
+            for requested_feature in requested_features:
+                api_requested_features.append(f"{table.name}:{requested_feature}")
+
+        req_body = json.dumps(
+            {
+                "features": api_requested_features,
+                "query": embedding,
+                "top_k": top_k,
+                "distance_metric": distance_metric,
+            }
+        )
+        return req_body
+
+    def _construct_online_documents_v2_api_json_request(
+        self,
+        table: FeatureView,
+        requested_features: Optional[List[str]],
+        embedding: Optional[List[float]],
+        top_k: int,
+        distance_metric: Optional[str] = None,
+        query_string: Optional[str] = None,
+        api_version: Optional[int] = 1,
```

Comment:
default api_version 2 ?

Reply:
I havent tested version 2 in depth for all online stores. BTW Do we have v2 implemented for all stores and confirm stability ?

If not, I think we can go with v1 as stable with opt-in to v2.